app/oxalis/security/CombinedAuthenticatorDAO.scala

```diff
@@ -0,0 +1,99 @@
+
+import javax.inject.Inject
+
+import com.scalableminds.util.tools.{Fox, FoxImplicits}
+import controllers.routes
+import models.user.{MongoUserTokenDao, User, UserService}
+import net.liftweb.common.{Empty, Full}
+import play.api.Play.current
+import play.api.libs.concurrent.Execution.Implicits._
+import play.api.libs.json.Json._
+import com.mohiva.play.silhouette.api.{Environment, LoginInfo, SecuredErrorHandler, Silhouette}
+import com.mohiva.play.silhouette.impl.authenticators.{BearerTokenAuthenticator, _}
+import com.mohiva.play.silhouette.impl.daos.AuthenticatorDAO
+import com.scalableminds.util.reactivemongo.{DBAccessContext, GlobalAccessContext}
+import models.basics.SecuredBaseDAO
+import org.joda.time.DateTime
+import play.api.libs.json._
+import play.modules.reactivemongo.ReactiveMongoApi
+import reactivemongo.play.json.collection.JSONCollection
+import play.api.libs.functional.syntax._
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.duration._
+import play.api.libs.json._
+import play.api.libs.json.Reads._
+import play.api.libs.json.Writes._
+import reactivemongo.api.commands.WriteResult
+import reactivemongo.bson.BSONObjectID
+import net.ceedubs.ficus.Ficus._
+import net.ceedubs.ficus.readers.ArbitraryTypeReader._
+import play.api.libs.json.Json
+/**
+  * Created by robert on 14.11.17.
+  */
+class CombinedAuthenticatorDAO extends AuthenticatorDAO[CombinedAuthenticator] with SecuredBaseDAO[CombinedAuthenticator]{
+  override def collectionName: String = "combinedAuthenticators"
+
+  implicit object FiniteDurationFormat extends Format[FiniteDuration] {
+    def reads(json: JsValue): JsResult[FiniteDuration] = LongReads.reads(json).map(_.seconds)
+    def writes(o: FiniteDuration): JsValue = LongWrites.writes(o.toSeconds)
+  }
+
+  implicit def optionFormat[T: Format]: Format[Option[T]] = new Format[Option[T]]{
+    override def reads(json: JsValue): JsResult[Option[T]] = json.validateOpt[T]
+
+    override def writes(o: Option[T]): JsValue = o match {
+      case Some(t) ⇒ implicitly[Writes[T]].writes(t)
+      case None ⇒ JsNull
+    }
+  }
+
+  override protected implicit def formatter: OFormat[CombinedAuthenticator] = JsonUtil.oFormat(Json.format[CombinedAuthenticator])
+
+  override def findOne[V](attribute: String, value: V)(implicit w: Writes[V], ctx: DBAccessContext): Fox[CombinedAuthenticator] = findOne(Json.obj(attribute -> w.writes(value)))
+
+  override def find(id: String): Future[Option[CombinedAuthenticator]] = findOne("id", id)(implicitly[Writes[String]],GlobalAccessContext).futureBox.map(box => box.toOption)
+
+  //adds the new token (and removes the old one)
+  override def add(authenticator: CombinedAuthenticator): Future[CombinedAuthenticator] = for{
+    maybeOldAuthenticator <- findByLoginInfo(authenticator.loginInfo)
+    _ <- insert(authenticator)(GlobalAccessContext).futureBox
+  }yield{
+    if(maybeOldAuthenticator.isDefined) remove(maybeOldAuthenticator.get.id)
```

Comment:
is it a scala Option? then I think this line could be expressed in maybeOldAuthenticator.map(remove(_.id))

Reply:
Yes it is. I changed it